app/src/main/java/kz/sozdik/main/MainActivity.kt

```diff
@@ -120,6 +121,15 @@ class MainActivity : BaseActivity() {
     }
 
     override fun onBackPressed() {
+        val mainFragment = supportFragmentManager.fragments.first()
```

Comment:
Для надежности лучше искать фрагмент вот так, т.к если логика сортировки фрагментов изменится в SDK, то это может сломать приложение
```suggestion
        val mainFragment = supportFragmentManager.findFragmentById(R.id.container)
```